Issue 402 create heatmap UI

PrayersonChauhan opened the pull request:
##  Heatmap Component

This PR adds the Heatmap component, which provides a unified interface for viewing evaluation counts in either *Month View* or *Week View*.

---

###  What It Does

- Renders a heatmap of evaluations grouped by:
  - *Month*: aggregated per month
  - *Week*: full calendar weeks (Monday–Sunday) based on selected month
- Allows toggling between month/week views
- Provides dropdowns to select year and month (month dropdown is shown only in Week View)

---

###  Props

| Prop         | Type           | Description                              |
|--------------|----------------|------------------------------------------|
| evaluations | Evaluation[] | A list of evaluations with dueDate values |

---

###  Usage Example

tsx
import Heatmap from './Heatmap';

<Heatmap evaluations={sampleEvaluations} />


The evaluations prop should contain items with dueDate fields. Internally, Heatmap passes the appropriate filtered props to either MonthView or WeekView.

----------

###  Related Components

- MonthView: displays evaluation counts by month
    
- WeekView: displays counts for all weeks covering a selected month

<img width="1264" height="952" alt="Screenshot 2025-07-16 183002" src="https://github.com/user-attachments/assets/f2a9db7b-6cab-41c6-ab55-64d1ca9de792" />
<img width="1264" height="952" alt="Screenshot 2025-07-16 183021" src="https://github.com/user-attachments/assets/3b5f0f1d-9114-42cd-a239-90dc3244a98a" />

achow-conestogac commented on src/App.tsx:
@SheetalRana10 and @UrviSurti3413 

since your task is similar, i wanted to tag you here as well.

achow-conestogac replied:
you need to address this here.

the heatmap should not be implemented here as it only pertains to Evaluations.

for this PR, you should not require any changes on the `App.tsx`.  the heat map should be on just on the Evaluation card.

PrayersonChauhan commented:
The Evaluation Dashboard code is still pending approval. During yesterday’s meeting, you suggested modifying the existing file rather than creating a new one. However, if we wait until the dashboard code is approved—which may not happen until Friday evening—there might not be enough time for our code to go through the review process.

As a temporary solution, we’ve added a link in the App. Would you recommend that we wait for the dashboard code to be approved first, or would it be acceptable to create a new file with the same name so it can be easily merged later?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Andy Chow ***@***.***>
Sent: Thursday, July 17, 2025 12:03:39 PM
To: conestogac-acsit/SENG8130-Software-Quality-Applications-Lab ***@***.***>
Cc: Prayerson Karneliusbhai Chauhan ***@***.***>; Author ***@***.***>
Subject: Re: [conestogac-acsit/SENG8130-Software-Quality-Applications-Lab] Issue 402 create heatmap UI (PR #568)

Caution: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe. Please send any questions or concerns to ***@***.***


@achow-conestogac commented on this pull request.

________________________________

On src/App.tsx<https://github.com/conestogac-acsit/SENG8130-Software-Quality-Applications-Lab/pull/568#discussion_r2213747350>:

@SheetalRana10<https://github.com/SheetalRana10> and @UrviSurti3413<https://github.com/UrviSurti3413>

since your task is similar, i wanted to tag you here as well.

—
Reply to this email directly, view it on GitHub<https://github.com/conestogac-acsit/SENG8130-Software-Quality-Applications-Lab/pull/568#discussion_r2213747350>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BSHWLWXXF27AD2RPO5IVGOT3I7CNXAVCNFSM6AAAAACBWFHRC6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTAMZQGE2TEMJQGI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

________________________________
Conestoga Official Disclaimer: This email and any attachments are for the sole use of the intended recipients and may be privileged or confidential. Any distribution, printing or other use by anyone else is prohibited. If you are not an intended recipient, please contact the sender immediately, and permanently delete this email and attachments.

PrayersonChauhan commented:
The Evaluation Dashboard has been approved and merged into the main branch, so we’ve updated our code accordingly. We've added the link within the Evaluation Dashboard. Please review our latest code. Thank you!

achow-conestogac reviewed (commented):
There's a bunch of errors when I run the dev environment. Please fix it.

PrayersonChauhan commented:
> There's a bunch of errors when I run the dev environment. Please fix it.

I have added placeholder for the weekview and monthview components to avoid the errors. Kindly review it.

achow-conestogac commented on src/Evaluation/Heatmap/Heatmap.tsx:
```diff
@@ -0,0 +1,60 @@
+          </label>
+        )}
+
+        <button
```
use the `Button` component from the component folder please.

PrayersonChauhan replied:
I replace the button using Button component. Please review it. Thank you.

achow-conestogac reviewed (approved):
please resolve all discussions before merging.
